File: c++/src/pixel.cpp
#include "pixel.hpp"

using namespace std;

/**
  Constructor of Pixel
*/
Pixel::Pixel() {}

/**
    Constructor of Pixel

    @param xVal : x coordinate of the pixel
    @param yVal : y coordinate of the pixel
    @param labelVal : label of the segment at pixel
    @param scoreVal : score of label at pixel
    @param keyVal : Unique identifier corresponding to point (x, y)
*/
Pixel::Pixel(int xVal, int yVal, int labelVal, float scoreVal, int keyVal) {
  x = xVal;
  y = yVal;
  label = labelVal;
  score = scoreVal;
  key = keyVal;
}


/**
  Destructor of Pixel
*/
Pixel::~Pixel(){}

File: python/pixel.py
class Pixel:
    """Class representing image segmentation pixel. Contains label and score.

    Parameters:
    -----------
    y: int
        y coordinate
    x: int
        x coordinate
    label: int
        segment label
    score: float
        label confidence
    """

    def __init__(self, y, x, label, score):
        self.y = y
        self.x = x
        self.label = label
        self.score = score
